xdsl/parser/core.py

```diff
@@ -811,11 +811,12 @@ def resolve_operands(
         and checks that the type is consistent.
 
         If the length of args and input_types does not match, an error is raised at
-        the the location error_pos.
+        the location error_pos.
         """
-        if len(args) != len(input_types):
+        length = len(list(input_types))
```

Comment:
I have to say, that feels like a hack.
Here, the method expects a `Sequence` and an `Iterable`, but checks the length of both.
I feel that having `Sequence` for both would make more sense, and we would just need to add `.data` during the function call.

Reply:
That works for me as well.

Reply:
It looks like the line stayed in. Could you please remove the temporary list allocation?